addons/im_livechat/static/src/legacy/widgets/public_livechat_view/public_livechat_view.xml

```diff
@@ -10,7 +10,7 @@
         @param {Array} [options.domain=[]] the domain to restrict messages on the thread.
     -->
     <t t-name="im_livechat.legacy.mail.widget.Thread">
-        <t t-if="thread.widget.hasMessages()">
+        <t t-if="widget.messaging.publicLivechatGlobal.publicLivechat.widget.hasMessages()">
```

Comment:
Is `widget` different from `widget.messaging.publicLivechatGlobal.publicLivechat.widget`? If so, is there a reason not to define `widget` to be `widget.messaging.publicLivechatGlobal.publicLivechat.widget` directly?

Reply:
The convention in old code is to access context of the widget with `widget`. OWL merges template context with component, hence why we don't need `comp` or something.

`widget !== widget.messaging.publicLivechatGlobal.publicLivechat.widget`